src/controllers/deleteProduct.controller.js

```diff
@@ -0,0 +1,32 @@
+import { Product } from '../Database/models';
+
+const deleteProduct = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const userId = req.user.id;
+    const product = await Product.findOne({
+      where: { id, userId },
```

Comment:
- why do we have to ask for the userId?
- Consider the case where a merchant different from the one who created the product is trying to delete the product.

Reply:
Because a merchant is supposed to delete their own product, they are not allowed to delete products belonging to others.

Reply:
There should be a condition to check if the logged in merchant is the one who wants to delete the product